jwst/tweakreg/wcsimage.py

```diff
@@ -208,35 +209,43 @@ def set_correction(self, matrix=[[1, 0], [0, 1]], shift=[0, 0]):
         self._update_transformations()
 
     def _check_wcs_structure(self, wcs):
+        valid = True
+        message = ""
         if wcs is None or wcs.pipeline is None:
-            return False
+            message = "WCS is None."
+            valid = False
 
-        frms = [f[0] for f in wcs.pipeline]
+        frms = wcs.available_frames
         nframes = len(frms)
         if nframes < 3:
-            return False
+            message = "There are more than 3 frames in this WCS pipeline."
```

Comment:
It should be the opposite: `"There are fewer than 3 frames in the WCS pipeline."`